conjure-go-client/httpclient/metrics.go

```diff
@@ -222,3 +231,25 @@ func (m *metricsWrappedConn) Close() error {
 	m.counter.Dec(1)
 	return m.Conn.Close()
 }
+
+func isTimeoutError(respErr error) bool {
+	if respErr == nil {
+		return false
+	}
+	rootErr := werror.RootCause(respErr)
+	if rootErr == nil {
+		return false
+	}
+
+	if nerr, ok := rootErr.(net.Error); ok && nerr.Timeout() {
```

Comment:
Should we also check for `nerr.Temporary()`?

Reply:
So from my reading of the various stdlib impls, timeout is mostly a subset of temporary (i.e. timeouts are temp) and where temporary is true but timeout is false I think we would want `other` to indicate it wasn't a timing problem

Reply:
cool that's my understanding as well